Missing header in `subsys/shell/modules/kernel_service/heap.c`

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->

The new file `subsys/shell/modules/kernel_service/heap.c` added in https://github.com/zephyrproject-rtos/zephyr/pull/77668 is missing `#include "kernel_shell.h"`, resulting in a build failure when this file is included:

```
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:13:41: warning: 'struct shell' declared inside parameter list will not be visible outside of this definition or declaration
   13 | static int cmd_kernel_heap(const struct shell *sh, size_t argc, char **argv)
      |                                         ^~~~~
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c: In function 'cmd_kernel_heap':
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:23:17: warning: implicit declaration of function 'shell_error' [-Wimplicit-function-declaration]
   23 |                 shell_error(sh, "Failed to read kernel system heap statistics (err %d)", err);
      |                 ^~~~~~~~~~~
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:24:25: error: 'ENOEXEC' undeclared (first use in this function)
   24 |                 return -ENOEXEC;
      |                         ^~~~~~~
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:24:25: note: each undeclared identifier is reported only once for each function it appears in
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:27:9: warning: implicit declaration of function 'shell_print' [-Wimplicit-function-declaration]
   27 |         shell_print(sh, "free:           %zu", stats.free_bytes);
      |         ^~~~~~~~~~~
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c: At top level:
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:34:21: error: expected ')' before '(' token
   34 | KERNEL_CMD_ADD(heap, NULL, "System heap usage statistics.", cmd_kernel_heap);
      |                     ^
      |                     )
/Users/gminnehan/projects/zephyr-rtos-workspace/zephyr/subsys/shell/modules/kernel_service/heap.c:13:12: warning: 'cmd_kernel_heap' defined but not used [-Wunused-function]
   13 | static int cmd_kernel_heap(const struct shell *sh, size_t argc, char **argv)
      |            ^~~~~~~~~~~~~~~
```

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->

Build a sample with shell enabled, heap > 0, and heap runtime stats enabled:

```
west build -b qemu_cortex_m3 zephyr/samples/hello_world -- -DCONFIG_SYS_HEAP_RUNTIME_STATS=y -DCONFIG_SHELL=y -DCONFIG_HEAP_MEM_POOL_SIZE=30000
```

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

Successful build with the above settings. Additionally, this looks to be missing coverage in CI. 

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->

Annoyance, currently a similar sample app used in CI tests is failing to build.

**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->

N/A

**Environment (please complete the following information):**

- OS: MacOS
- Toolchain: Zephyr SDK
- Testing on Zephyr latest. At the time of writing, tests were run on 9d9089edd09

**Additional context**
<!--
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
-->

N/A

--- a/subsys/shell/modules/kernel_service/heap.c
+++ b/subsys/shell/modules/kernel_service/heap.c
@@ -6,6 +6,8 @@
  */
 
 #if K_HEAP_MEM_POOL_SIZE > 0
+#include "kernel_shell.h"
+
 #include <zephyr/sys/sys_heap.h>
 
 extern struct sys_heap _system_heap;

